src/components/atoms/Icon/Icon.tsx

```diff
@@ -1,11 +1,34 @@
 import React from 'react';
+import styled from 'styled-components';
 import { FontAwesomeIcon, FontAwesomeIconProps } from '@fortawesome/react-fontawesome';
 import { IconDefinition } from '@fortawesome/fontawesome-common-types';
 
+import { Colors } from '../../../constants/colors';
+import { spacing } from '../../../constants/spacing';
+
 interface IconProps extends Omit<FontAwesomeIconProps, 'icon' | 'border'> {
   variant: IconDefinition;
+  bgColor?: Colors[keyof Colors];
 }
 
-export default function Icon({ variant, ...rest }: IconProps) {
-  return <FontAwesomeIcon {...rest} icon={variant} />;
+export const RoundBadge = styled.div<{ color: Colors[keyof Colors] }>`
```

Comment:
Do we want to allow data-automation support for the wrapping item?

Reply:
Adding data-automation will be added to the svg icon